### PR TITLE
Check members of class expression too

### LINKED ISSUE #4431: Class expressions not checked for missing properties
In 1.6, we're currently not catching when a class expression references a non-existent property.

For example, we don't error on the use of 'this.y' in the following example:

```TypeScript
class George extends class { reset() { return this.y; } } {
    constructor() {
        super();
    }
}
```

Our behavior here should match class declarations and warn on the use of the undeclared 'y'

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #4431

### CODE REVIEW

File: tests/baselines/reference/classExpression4.types [hunk 1]
```diff
@@ -1,22 +1,22 @@
 === tests/cases/conformance/classes/classExpressions/classExpression4.ts ===
 let C = class {
->C : typeof (Anonymous class)
+>C : any
```
Comment (sheetalkamat) [inline tests/baselines/reference/classExpression4.types]: @ahejlsberg @mhegazy is this correct? Or should we do something special for this case

Comment (sheetalkamat) [inline tests/baselines/reference/classExpression4.types]: It becomes any because C is accessed again as part of method resolution of class expression.

Comment (DanielRosenwasser) [inline tests/baselines/reference/classExpression4.types]: Weirdly enough, this logic was already done in `checkFunctionAndClassExpressionBodies`:

```TypeScript
case SyntaxKind.ClassExpression:
    forEach((<ClassExpression>node).members, checkSourceElement);
    break;
```

So the question is, why wasn't this ever doing its job?

Comment (sheetalkamat) [inline tests/baselines/reference/classExpression4.types]: Hmm. Let me look into that

**Comment by ahejlsberg** [comment]
I have put up a suggested simpler fix in #4501. Feel free to grab from it.

**Comment by sheetalkamat** [comment]
Closing since #4501 got this.

**Pull request closed without merging**